### PR TITLE
fix: refactor linea testnet implementation

### PR DESCRIPTION
**Pull request by VGau**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

- Refactored Linea testnet implementation to use the Infura rpc provider
- Updated Linea testnet rpc url to use Infura rpc url (https://linea-goerli.infura.io/v3/)
- Added new package patches to be compatible with Linea testnet
- Created a PR in metamask core to add linea networks support (https://github.com/MetaMask/core/pull/1423)

**Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/85494462/057ffa84-9522-45f7-b7c2-c9a83128f9a1

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by sethkfman** [comment]
@VGau Can you show test evidence that the new refactor is still working?

File: patches/@metamask+transaction-controller+4.0.0.patch
Comment (sethkfman) [inline patches/@metamask+transaction-controller+4.0.0.patch:1]: Have you opened PRs for the controllers? If not, please do and reference them in this pr.

Comment (VGau) [reply patches/@metamask+transaction-controller+4.0.0.patch:1]: As you know the MM team is currently refectoring the MM core package. @Gudahtt told me to only create patches for now. Maybe I misunderstood :) WDYT ?

**Comment by sethkfman** [comment]
@VGau are these ready for dev-review?

**Comment by VGau** [comment]
> @VGau are these ready for dev-review?

@sethkfman Yes this PR is ready for dev-review, as well as this one #6496 :) For your information, the PR #6496 is based on this branch. Maybe it is better to review the PR #6496 :)

File: patches/@metamask+transaction-controller+4.0.0.patch
Comment (Gudahtt) [reply patches/@metamask+transaction-controller+4.0.0.patch:1]: Sorry for the confusion; I meant that you should do both. We should fix this upstream in core, but core will probably take a couple of weeks at least to update, so you'll want to write a patch to get this in sooner.

File: app/components/Views/Settings/NetworksSettings/index.js [hunk 1]
```diff
@@ -238,7 +237,11 @@ class NetworksSettings extends PureComponent {
                 {!isCustomRPC &&
                   (image ? (
                     <ImageIcons
-                      image={network.toUpperCase()}
+                      image={
+                        network !== LINEA_GOERLI
+                          ? network.toUpperCase()
```
Comment (sethkfman) [inline app/components/Views/Settings/NetworksSettings/index.js:242]: Why is this check needed? Can't we use the network name?

File: patches/@metamask+transaction-controller+4.0.0.patch
Comment (sethkfman) [reply patches/@metamask+transaction-controller+4.0.0.patch:1]: @VGau once, you link the controllers PR and address the comment below the rest looks good.

Comment (VGau) [reply patches/@metamask+transaction-controller+4.0.0.patch:1]: @sethkfman @Gudahtt I created a PR in the metamask core repository. The link is in the description.

File: app/components/Views/Settings/NetworksSettings/index.js (hunk 1 above)
Comment (sethkfman) [reply app/components/Views/Settings/NetworksSettings/index.js:242]: can you change the image name?

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by sethkfman** [comment]
@cortisiko You will not see any functional difference. It should only need regression/e2e QA but I wanted you to confirm.

**Comment by cortisiko** [comment]
@VGau additional test cases:

- tap on LineaEth and ensure the app doesn't crash. We recently added charts to the token overview screen.
- ensure that you are unable to edit/delete the Linea testnet

**Review by cortisiko — APPROVED** [review]
✅ ✅  This PR looks good to me. Once CI is passed we can merge

**Comment by VGau** [comment]
> ✅ ✅ This PR looks good to me. Once CI is passed we can merge

Thanks @cortisiko  😄  The CI is failing because there is an error with SonarCloud. Is it possible to check this error ?

**Pull request merged**